### PR TITLE
 Parse pubDate in RSS 0.91

### PR DESCRIPTION
**Pull request by JampireX**
#414

### CODE REVIEW

File: rome/src/main/java/com/rometools/rome/feed/synd/impl/ConverterForRSS091Userland.java [hunk 1]
```diff
@@ -89,6 +76,11 @@ public void copyInto(final WireFeed feed, final SyndFeed syndFeed) {
                 creators.addAll(s);
             }
         }
+
+        final List<Item> items = channel.getItems();
+        if (items != null) {
+            syndFeed.setEntries(createSyndEntries(items, syndFeed.isPreservingWireFeed()));
+        }
```
Comment (mishako) [inline rome/src/main/java/com/rometools/rome/feed/synd/impl/ConverterForRSS091Userland.java]: Is this code necessary? Doesn't `ConverterForRSS091Userland` inherit it from `ConverterForRSS090`?

File: rome/src/main/java/com/rometools/rome/io/impl/RSS091UserlandParser.java [hunk 2]
```diff
@@ -148,6 +143,8 @@ protected WireFeed parseChannel(final Element rssRoot, final Locale locale) {
             channel.setSkipDays(skipDays);
         }
 
+        channel.setItems(parseItems(rssRoot, locale));
```
Comment (mishako) [inline rome/src/main/java/com/rometools/rome/io/impl/RSS091UserlandParser.java]: Same as my previous comment - should work without it?

File: rome/src/main/java/com/rometools/rome/io/impl/RSS091UserlandParser.java [hunk 3]
```diff
@@ -252,6 +249,25 @@ protected Element getTextInput(final Element rssRoot) {
...
+     * @param rssRoot the root element of the RSS document to parse for all items information.
+     * @return a list with all the parsed RSSItem beans.
+     */
+    protected List<Item> parseItems(final Element rssRoot, final Locale locale) {
+        final List<Item> items = new ArrayList<Item>();
+        for (final Element item : getItems(rssRoot)) {
+            items.add(parseItem(rssRoot, item, locale));
+        }
+        return items;
+    }
```
Comment (mishako) [inline rome/src/main/java/com/rometools/rome/io/impl/RSS091UserlandParser.java]: Same as previous comments.

**Comment by mishako** [comment]
Build failed:

```
Failed tests:
testPubDateDoesNotOverrideDublinCoreDate(com.rometools.rome.unittest.DublinCoreTest):
expected:<Thu Jan 01 00:00:01 UTC 1970> but was:<Thu Jan 01 00:00:00 UTC 1970>
```

File: rome/src/main/java/com/rometools/rome/io/impl/RSS091UserlandParser.java (hunk 2 above)
Comment (JampireX) [reply rome/src/main/java/com/rometools/rome/io/impl/RSS091UserlandParser.java]: Yes, removed.

File: rome/src/main/java/com/rometools/rome/feed/synd/impl/ConverterForRSS091Userland.java [hunk 4]
```diff
@@ -166,6 +166,11 @@ protected SyndEntry createSyndEntry(final Item item, final boolean preserveWireI
 
         syndEntry.setComments(item.getComments());
 
+        // DublinCoreTest will be failed without this row
+        // I don't have better solution
```
Comment (mishako) [inline rome/src/main/java/com/rometools/rome/feed/synd/impl/ConverterForRSS091Userland.java]: It looks like you didn't figure it out? I mean, why exactly the test fails and whether it means that something is broken for other use cases.

Comment (JampireX) [reply rome/src/main/java/com/rometools/rome/feed/synd/impl/ConverterForRSS091Userland.java]: It's little dirty solution for this problem but i think it's shouldn't broke anything else. This problem need more deep solution for understanding cross-version syntax

**Pull request merged**